api/src/main/java/io/opentelemetry/api/metrics/batch/BatchObserverContext.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.metrics.batch;
+
+import io.opentelemetry.api.common.Labels;
+import io.opentelemetry.api.metrics.Instrument;
+import io.opentelemetry.api.metrics.batch.AsynchronousBatchInstrument.Observation;
+import java.util.function.Consumer;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Util class for creating multiple instruments that can be atomically observed in a single
+ * callback.
+ *
+ * <p>This class is equivalent to individually calling {@code observe} on every {@link Instrument},
+ * but has the advantage of all recordings occurring atomically.
+ *
+ * <p>Example:
+ *
+ * <pre>{@code
+ * meter.newBatchObserver(
+ *     context -> {
+ *       DoubleSumBatchObserver timeObserver =
+ *           context.doubleSumObserverBuilder("time").setUnit("s").build();
+ *       LongSumBatchObserver distanceObserver =
+ *           context.longSumObserverBuilder("distance").setUnit("m").build();
+ *
+ *       context.registerCallback(
+ *           result ->
+ *               MyMeasure measure = MyMeasure.getdata();
+ *               result.observe(
+ *                   Labels.of("myKey", "myValue"),
+ *                   distanceObserver.observation(measure.getDistance()),
+ *                   timeObserver.observation(measure.getTime())));
```

Comment:
@jmacd What do you think about this API shape for BatchObserver registration? @kenfinnigan and I are thinking about proposing this as a clarification in the specs for this API.

Reply:
LGTM. If another `registerCallback`  happens later, it replaces the format callback, I imagine?

Reply:
I think it would be ideal if there was no `registerCallback` function and the callback is only passed during instrument creation. One design possibility would be

>  An async instrument creation function accepts a callback but returns nothing. Then the callback is invoked with the instrument as parameter. This prevents the API user from accidentally calling the observe() function outside the interval.

-- https://github.com/open-telemetry/opentelemetry-specification/pull/1030#issuecomment-707668179

Reply:
@Oberon00 I like the idea of preventing unintentional calling of `observe()`, but I wasn't able to come up with how this could work given the callback is taking multiple instruments?

Maybe I need more coffee this morning. Can you provide a pseudo-code example?

Reply:
Actually I don't think this would work as-is for the BatchObserver, only for the other instruments. But maybe we can still make it so that there is no registerCallback method?

Reply:
Maybe creation of the BatchObserver can accept a set of Labels and an array of instruments?

Something like:

```
meter.newBatchObserver(
    Labels.of("myKey", "myValue"),
    meter.doubleSumObserverBuilder("time").setUnit("s").build(),
    meter.longSumObserverBuilder("distance").setUnit("m").build()
)
```

Haven't confirmed this would "work" for the observe side, just a quick stab at what you're thinking